src/main/java/omero/gateway/Gateway.java

```diff
@@ -1701,4 +1724,31 @@ public void close() throws Exception {
         if (connected)
             disconnect();
     }
+
+    /**
+     * Wrapper holding a client and a flag if the client is hooked up
+     * to a already existing session.
+     */
+    private class SessionWrapper {
```

Comment:
Interesting. Does this actually mean you'd benefit from having the `sessionLogin` flag in `omero.client` itself?

Reply:
Well... yes, in the end it actually could be a useful property of the `omero.client` itself.